docs-vim.js

```diff
@@ -57,36 +71,91 @@ vim.normal_keydown = function (e) {
 	vim.num = "";
     }
 
-    if (e.key == "V" && e.shift) {
-        console.log("select");
+    return false;
+};
+
+// Called in visual mode.
+vim.visual_keydown = function (e) {
+    if (e.key.match(/F\d+/)) {
+        // Pass through any function keys.
+        return true;
+    }
+
+    if (e.key == "Escape") {
+	// Escape visual mode.
+        vim.switchToNormalMode();
+    }
+
+    vim.currentSequence += e.key;
+    if (vim.currentSequence == vim.keys.escapeSequence) {
+        e.preventDefault();
+        e.stopPropagation();
+
+        vim.switchToNormalMode();
+        return false;
+    }
+    if (vim.keys.escapeSequence.indexOf(vim.currentSequence) != 0) {
+        vim.currentSequence = e.key;
+    }
+
+    e.preventDefault();
+    e.stopPropagation();
+
+    var keyMap = { "Backspace": "ArrowLeft", "x": "Delete" };
+    keyMap[vim.keys.move[0]] = "ArrowLeft";
+    keyMap[vim.keys.move[1]] = "ArrowDown";
+    keyMap[vim.keys.move[2]] = "ArrowUp";
+    keyMap[vim.keys.move[3]] = "ArrowRight";
```

Comment:
I think this is duplicated with `normal_keydown`. Not super important to resolve in this PR, but we can do a code-deduplication pass at some point in the future and see if we can refactor stuff like this out.